docs/development/visualize/development-create-visualization.asciidoc

```diff
@@ -0,0 +1,404 @@
+
+  return VisFactory.createAngularVisualization({
+    name: 'my_new_vis',
+    title: 'My New Vis',
+    icon: 'my_icon',
+    description: 'Cool new chart',
+    editorController: MyEditorController,
+    editorConfig: { my: 'custom config' }
+  });
+}
+-----------
+
+[[development-visualization-request-handlers]]
+=== Visualization Request Handlers
+Request handler gets called when one of the following keys on AppState change:
+`vis`, `query`, `filters` or `uiState`. This sums up to pretty much every change to visualization. On top
+of that it will also get called on force refresh.
+
+By default visualizations will use courier request handler. They can also choose to use any of the other provided 
+request handlers. It's also possible to define your own request handler 
+(which you can then register to be used by other visualizations).
+
+[[development-default-request-handler]]
+==== courier request handler
+'courier' is the default request handler which works with the 'default' side bar editor.
+
+[[development-none-request-handler]]
+==== `none` request handler
+Using 'none' as your request handles means your visualization does not require any data to be requested.
+
+[[development-custom-request-handler]]
+==== custom request handler
+You can define your custom request handler by providing a function with the following definition:
+`function (vis, appState, uiState, searchSource) { ... }`
+
+Function must return a promise, which should get resolved with new data that will be passed to responseHandler.
+
+It's up to function to decide when it wants to issue a new request or return previous data 
+(if none of the objects relevant to the request handler changed).
+
+["source","js"]
+-----------
+import { VisFactoryProvider } from 'ui/vis/vis_factory';
+
+const myRequestHandler = (vis, appState, uiState, searchSource) => {
+  const shouldQuery = () => {
+     // check if any of the relevant objects changed
+     // if not we should just return the old data
+  };
+  
+  return new Promise((resolve, reject) => {
+     if (shouldQuery()) {
+        // get your data
+        // resolve(myNewData);
+     } else {
+       resolve(myOldData);
+     }
+  });
+};
+
+export default function MyNewVisType(Private) {
+  const VisFactory = Private(VisFactoryProvider);
+
+  return VisFactory.createAngularVisualization({
+    name: 'my_new_vis',
+    title: 'My New Vis',
+    icon: 'my_icon',
+    description: 'Cool new chart',
+    requestHandler: myRequestHandler
+  });
+}
+-----------
+
+[[development-visualization-response-handlers]]
+=== Visualization Response Handlers
+Response handler will receive the data from request handler and an instance of Vis object. 
+Its job is to convert the data to a format visualization can use. By default 'none' request handler is used 
+which just passes the data it received back. The data object will then be passed to visualization.
+
+[[development-default-response-handler]]
+==== default response handler
```

Comment:
Why is this called the `default` response handler when the above paragraph states that "By default 'none' request handler is used". Maybe this should be called the 'tabular' response handler?

Reply:
actually the default should be default, none should be possible but unadvised.
with 'none' visualization gets pure es data in which:
- is hard to work with
- might change in the future
- locks visualization to being used with es responses only